snapcraft/main.py

```diff
@@ -81,25 +86,27 @@
     'upload',
 ]
 
-try:
-    version = pkg_resources.require('snapcraft')[0].version
-except pkg_resources.DistributionNotFound:
-    version = 'devel'
+
+def _version():
+    try:
+        return pkg_resources.require('snapcraft')[0].version
+    except pkg_resources.DistributionNotFound:
+        return 'devel'
 
 
 def main():
     log.configure()
-    args = docopt(__doc__,
-                  version=version,
-                  options_first=True)
-    if args['COMMAND'] not in _VALID_COMMANDS:
-        sys.exit('Command {!r} was not recognized'.format(args['COMMAND']))
+    args = docopt(__doc__, version=_version(), options_first=True)
+
+    cmd = args['COMMAND'] or 'snap'
+    if cmd not in _VALID_COMMANDS:
+        sys.exit('Command {!r} was not recognized'.format(cmd))
 
     try:
-        commands.load(args['COMMAND']).main(argv=args['ARGS'])
+        commands.load(cmd).main(argv=args['ARGS'])
     except Exception as e:
         sys.exit(textwrap.fill(str(e)))
 
 
-if __name__ == '__main__':
-    main()
+if __name__ == '__main__':  # pragma: no cover
+    main()                  # pragma: no cover
```

Comment:
integration tests do cover this, but it is meaningless (and unittests don't get 100% either :wink:)